source/common/filter/tcp_proxy.h

```diff
@@ -38,13 +39,31 @@ class TcpProxyConfig {
   TcpProxyConfig(const Json::Object& config, Upstream::ClusterManager& cluster_manager,
                  Stats::Store& stats_store);
 
-  const std::string& clusterName() { return cluster_name_; }
+  /**
+   * Find out which cluster an upstream connection should be opened to.
+   * @param connection supplies the parameters of the downstream connection for
+   * which the proxy needs to open the corresponding upstream
+   * @return the cluster name to be used for the upstream connection.
+     If no route applies, returns the empty string.
+   */
+  const std::string& getClusterForConnection(Network::Connection& connection);
+
   const TcpProxyStats& stats() { return stats_; }
 
 private:
+  struct Route {
```

Comment:
@rshriram I agree that ultimately we will want to make this more complicate, and probably use RDS, etc., but for now in the interest of time I think it's OK to keep this here since it's so simple. We can refactor it out in a future change.